### PR TITLE
[PWA] Order confirmation page

### PR DESCRIPTION
**Pull request by Kronnox**
Part of #206 

Test URLs:
- Before: https://main--maidenform--hlxsites.hlx.live/
- After: https://issue-206--maidenform--hlxsites.hlx.live/

(No direct link present)

### CODE REVIEW

File: src/pages/checkout/confirmation/orderConfirmationPage.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                    defaultMessage={'Thank you'}
+                />
+            </h1>
+            <p className={classes.info}>
+                <FormattedMessage
+                    id={'checkoutConfirmationPage.orderInfo'}
+                    defaultMessage={'Order number {orderNumberLink} has been placed.'}
+                    values={{
+                        orderNumberLink:
+                            <Link to={`/customer/order-history?order=${orderNumber}`} className={classes.orderNumber}>
```
[the remark this answers is not in the document]
Comment (herzog31) [reply src/pages/checkout/confirmation/orderConfirmationPage.js:27]: Also be aware, the link has to point to `order.id` while the `order.orderNumber` should be displayed.

[the remark this answers is not in the document]
Comment (herzog31) [reply src/pages/checkout/confirmation/orderConfirmationPage.js:27]: What happens if the user is not logged in, but the whole checkout was done as a guest user? Then this link would not lead anywhere.

File: src/pages/checkout/confirmation/orderConfirmationPage.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import { FormattedMessage } from 'react-intl';
+import { shape, string } from 'prop-types';
+
+import defaultClasses from './orderConfirmationPage.module.css';
+import {useStyle} from "@magento/venia-ui/lib/classify";
```
Comment (herzog31) [inline src/pages/checkout/confirmation/orderConfirmationPage.js:6]:
```suggestion
import useStyle from '@magento/peregrine/lib/util/shallowMerge';
```

File: src/pages/checkout/confirmation/orderConfirmationPage.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Kronnox) [reply src/pages/checkout/confirmation/orderConfirmationPage.js:27]: > Also be aware, the link has to point to order.id while the order.orderNumber should be displayed.

You're right. We'll have to see on how to implement this as soon as we are integrating with the real checkout process. I'll add a TODO for the meantime.

[the remark this answers is not in the document]
Comment (Kronnox) [reply src/pages/checkout/confirmation/orderConfirmationPage.js:27]: > What happens if the user is not logged in, but the whole checkout was done as a guest user? Then this link would not lead anywhere.

@herzog31 Valid question. AFAIK there isn't a way to display an order without being signed in? Should we just remove the link in these cases?

[the remark this answers is not in the document]
Comment (herzog31) [reply src/pages/checkout/confirmation/orderConfirmationPage.js:27]: > Valid question. AFAIK there isn't a way to display an order without being signed in? Should we just remove the link in these cases?

Sounds good 👍

File: src/pages/checkout/confirmation/orderConfirmationPage.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    const classes = useStyle(defaultClasses, props.classes);
+    const { orderNumber } = props;
+
+    const [commerceScript, setCommerceScript] = useState(null);
+    useEffect(() => {
+      import(/* webpackIgnore: true */ '/scripts/commerce.js').then(instance => setCommerceScript(instance));
+    });
+    const [ratingState, setRatingState] = useState(0); // 0 = ready; 1 = pending; 2 = success
+
+    const handleRating = (rating) => {
```
Comment (herzog31) [inline src/pages/checkout/confirmation/orderConfirmationPage.js:22]: You can simplify the method like this:

```javascript
const handleRating = async (rating) => {
        setRatingState(1);
        // You can fetch the commerce script dynamically, no need to store this in the state. The browser optimizes enough to not load this script multiple times.
        const { getMagentoStorefrontEvents } = await import(/* webpackIgnore: true */ '/scripts/commerce.js'); 

        getMagentoStorefrontEvents((mse) => {
            if (!mse || ratingState > 1) { // You had ratingState > 0 before, but you should actually on skip if the rating was already successfully stored.
                return;
            }
            window.adobeDataLayer = window.adobeDataLayer || [];
            window.adobeDataLayer.push({
            "order-rating": {
                "orderNumber": orderNumber,
                "rating": rating
            }
            });
            setRatingState(2);
        });
    }
```

**Review by herzog31 — APPROVED** [review]
Looks great 👍

**Comment by Kronnox** [comment]
@herzog31 Should we merge this right now, to get the changes in and do a separate PR for integrating it as soon as the new checkout is merged?

**Comment by herzog31** [comment]
@Kronnox yes, please merge. Branches/PRs that are open for a long time tend to have lots of merge conflicts, so better get it merged early :)

**Pull request merged**